packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -9,215 +9,281 @@ import {
+    volume: Number(process.env["THRESHOLD_VOLUME"]) || 500_000,
   };
 }
 
-async function processProposal(
+const blocksPerSecond = POLYGON_BLOCKS_PER_HOUR / 3_600;
+
+function outcomeIndexes(
+  isSportsMarket: boolean,
   proposal: OptimisticPriceRequest,
-  params: MonitoringParams,
-  logger: typeof Logger,
-  version: "v1" | "v2"
-) {
-  const isSportsMarket = proposal.requester === params.ctfSportsOracleAddress;
-  const questionID = calculatePolymarketQuestionID(proposal.ancillaryData);
+  sportsMarket?: Market
+): { winner: number; loser: number; scores: [ethers.BigNumber, ethers.BigNumber]; mvq?: MultipleValuesQuery } {
+  if (isSportsMarket) {
+    if (isUnresolvable(proposal.proposedPrice))
+      return { winner: -1, loser: -1, scores: [ethers.constants.Zero, ethers.constants.Zero] };
 
-  let markets;
-  try {
-    markets = await getPolymarketMarketInformation(logger, params, questionID);
-  } catch (error) {
-    // Check if this is the specific "No market found" error for 3rd party proposals
-    if (error instanceof Error && error.message.includes(`No market found for question ID: ${questionID}`)) {
-      // Apply 3rd party proposal filtering logic
-      const shouldIgnore = await shouldIgnoreThirdPartyProposal(params, proposal, version);
-
-      if (shouldIgnore) {
-        // Ignore this proposal - log for debugging but don't alert
-        logger.info({
-          at: "PolymarketMonitor",
-          message: "Ignoring 3rd party Polymarket proposal based on filtering criteria",
-          questionID,
-          proposalHash: proposal.proposalHash,
-          requester: proposal.requester,
-        });
-        return { notified: false, notifiedProposal: null };
-      }
-      // If <2 criteria met, let the error bubble up to trigger alert
-    }
-    // Re-throw the error to maintain existing behavior for other error types
-    throw error;
+    const scores: [ethers.BigNumber, ethers.BigNumber] = [
+      decodeMultipleQueryPriceAtIndex(proposal.proposedPrice, 0),
+      decodeMultipleQueryPriceAtIndex(proposal.proposedPrice, 1),
+    ];
+    const payouts = getSportsPayouts(sportsMarket!, proposal.proposedPrice);
+    if (payouts[0] === payouts[1]) return { winner: -1, loser: -1, scores };
+
+    const winner = payouts[0] === 1 ? 0 : 1;
+    return {
+      winner,
+      loser: 1 - winner,
+      scores,
+      mvq: decodeMultipleValuesQuery(tryHexToUtf8String(proposal.ancillaryData)),
+    };
   }
 
-  for (const marketInfo of markets) {
-    let scores: [ethers.BigNumber, ethers.BigNumber] = [ethers.BigNumber.from(0), ethers.BigNumber.from(0)];
-    let multipleValuesQuery: MultipleValuesQuery | undefined;
-    let winnerOutcome: number;
-    let loserOutcome: number;
+  // Non-sports market logic.
+  // Ignore unresolvable prices where the orderbook doesn't provide relevant information.
+  if (!proposal.proposedPrice.eq(ethers.BigNumber.from(0)) && !proposal.proposedPrice.eq(ONE_SCALED)) {
```

Comment:
nit: does this mean we treat all requests that are not 1 and not 0 as 50/50?

Does PM also interpret them that way onchain? Naively, I would assume we would _always_ want to alert for a proposal that isn't a valid value (i.e. 0, 1, or 0.5).

This is a very minor point, and should not block us from merging this!

Thoughts?